spec/registry.dtd

```diff
@@ -0,0 +1,33 @@
+<!ELEMENT registry (function*|pattern*)>
+
+<!ELEMENT function (description|signature+)>
+<!ATTLIST function name NMTOKEN #REQUIRED>
+
+<!ELEMENT description (#PCDATA)>
+
+<!ELEMENT pattern EMPTY>
+<!ATTLIST pattern id ID #REQUIRED>
+<!ATTLIST pattern regex CDATA #REQUIRED>
+
+<!ELEMENT signature (input*|param*|match*)>
+<!ATTLIST signature type (match|format) #REQUIRED>
```

Comment:
Is there a particular reason why it'd be better to have `<signature type="match">` and `<signature type="format">` rather than e.g. `<match-signature>` and `<format-signature>`?

Separating the two would allow for attributes like `position` and child elements like `<match>` to only be valid for one element, and not the other.

Reply:
That's a good point. Done in 39f13a99db4f0391ebabd8534dbf54e190c6c827, thanks!